app/services/api/v3/import/importer.rb

```diff
@@ -80,6 +81,51 @@ def import
                 yellow_table_stats
               )
           end
+          destroy_widows
+        end
+
+        # An example of how a widow can be created by the import process:
+        # 1. backup xxx_inds/quals/quants and xxx_attributes (yellow)
+        # 2. import and replace inds/quals/quants (blue)
+        # 3. restore xxx_attributes (note: this restores all the backed up
+        #    records, because at this point we don't know if any of them are
+        #    no longer needed)
+        # 4. restore xxx_inds/quals/quants (note: this only restores backed up
+        #    records with a match in both inds/quals/quants and xxx_attributes)
+        # 5. at this point it is possible to have xxx_attributes which are not
+        # referenced anywhere. These widows should be deleted.
+        def destroy_widows
+          TABLES_TO_CHECK_FOR_WIDOWS.each do |table_class|
+            cnt_before = table_class.count
+            table_class.destroy_widows
+            cnt_after = table_class.count
+            next unless cnt_after != cnt_before
+
+            update_yellow_table_stats_after(table_class, cnt_after)
+          end
+        end
+
+        def update_yellow_table_stats_after(table_class, cnt_after)
+          table_name = table_class.table_name
+          blue_table_name = ALL_TABLES.find do |table|
```

Comment:
I think that would not return table_name but the whole hash, the table name still will be in `table[:table_class].table_name`

Reply:
That could be then shorten to 
```
blue_table_name = ALL_TABLES.find { |t| { t[:yellow_tables]&.include?(table_class) }&.dig(:table_class)&.table_name
```
not exactly in one line if too long